homeassistant/components/electric_kiwi/coordinator.py

```diff
@@ -0,0 +1,80 @@
+"""Electric Kiwi coordinators."""
+from datetime import timedelta
+import logging
+
+import async_timeout
+from electrickiwi_api import ElectricKiwiApi
+from electrickiwi_api.exceptions import ApiException, AuthException
+from electrickiwi_api.model import Hop, HopIntervals
+
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+_LOGGER = logging.getLogger(__name__)
+
+ACCOUNT_SCAN_INTERVAL = timedelta(hours=6)
+HOP_SCAN_INTERVAL = timedelta(hours=2)
+
+
+class ElectricKiwiHOPDataCoordinator(DataUpdateCoordinator):
+    """ElectricKiwi Data object."""
+
+    def __init__(self, hass: HomeAssistant, ek_api: ElectricKiwiApi) -> None:
+        """Initialize ElectricKiwiAccountDataCoordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            # Name of the data. For logging purposes.
+            name="Electric Kiwi HOP Data",
+            # Polling interval. Will only be polled if there are subscribers.
+            update_interval=HOP_SCAN_INTERVAL,
+        )
+        self._ek_api = ek_api
+        self.hop_intervals: HopIntervals = None
+
+    def get_hop_options(self) -> dict[str, int]:
+        """Get the hop interval options for selection."""
+        return {
+            f"{v.start_time} - {v.end_time}": k
+            for k, v in self.hop_intervals.intervals.items()
+        }
+
+    def get_selected_hop(self) -> Hop:
+        """Get currently selected hop."""
+        return self.data
```

Comment:
This method seems unneeded, as it returns self.data, just use self.data in that case?

Reply:
removed